kotlin-reference-server/src/main/kotlin/org/stellar/reference/event/processor/Sep6EventProcessor.kt

```diff
@@ -0,0 +1,204 @@
+package org.stellar.reference.event.processor
+
+import java.lang.RuntimeException
+import java.time.Instant
+import kotlinx.coroutines.runBlocking
+import org.stellar.anchor.api.callback.GetCustomerRequest
+import org.stellar.anchor.api.event.AnchorEvent
+import org.stellar.anchor.api.platform.PatchTransactionRequest
+import org.stellar.anchor.api.platform.PatchTransactionsRequest
+import org.stellar.anchor.api.platform.PlatformTransactionData
+import org.stellar.anchor.api.sep.SepTransactionStatus
+import org.stellar.anchor.api.shared.Amount
+import org.stellar.anchor.api.shared.InstructionField
+import org.stellar.anchor.api.shared.StellarPayment
+import org.stellar.anchor.api.shared.StellarTransaction
+import org.stellar.reference.callbacks.customer.CustomerService
+import org.stellar.reference.client.PlatformClient
+import org.stellar.reference.data.Config
+import org.stellar.reference.log
+import org.stellar.sdk.Asset
+import org.stellar.sdk.KeyPair
+import org.stellar.sdk.Network
+import org.stellar.sdk.PaymentOperation
+import org.stellar.sdk.Server
+import org.stellar.sdk.TransactionBuilder
+
+class Sep6EventProcessor(
+  private val config: Config,
+  private val server: Server,
+  private val platformClient: PlatformClient,
+  private val customerService: CustomerService,
+  private val activeTransactionStore: ActiveTransactionStore
+) : SepAnchorEventProcessor {
+  override fun onQuoteCreated(event: AnchorEvent) {
+    TODO("Not yet implemented")
+  }
+
+  override fun onTransactionCreated(event: AnchorEvent) {
+    when (val kind = event.transaction.kind) {
+      PlatformTransactionData.Kind.DEPOSIT -> onDepositTransactionCreated(event)
+      PlatformTransactionData.Kind.WITHDRAWAL -> TODO("Withdrawals not yet supported")
+      else -> {
+        log.warn("Received transaction created event with unsupported kind: $kind")
+      }
+    }
+  }
+
+  private fun onDepositTransactionCreated(event: AnchorEvent) {
+    if (event.transaction.status != SepTransactionStatus.INCOMPLETE) {
+      log.warn(
+        "Received deposit transaction created event with unsupported status: ${event.transaction.status}"
+      )
+      return
+    }
+    val customer =
+      customerService.getCustomer(
+        GetCustomerRequest.builder().account(event.transaction.destinationAccount).build()
+      )
+    runBlocking {
+      patchTransaction(
+        PlatformTransactionData.builder()
+          .id(event.transaction.id)
+          .status(SepTransactionStatus.PENDING_ANCHOR)
+          .build()
+      )
+    }
+    activeTransactionStore.addTransaction(customer.id, event.transaction.id)
```

Comment:
What is activeTransactionStore for? Can't we always get transaction from AP?

Reply:
Yeah, that would work too. If it's all the same to you, let's keep the current implementation to minimize additional changes.

Reply:
I think we can remove it later to keep it cleaner. If anybody look at the code they would think they should store transaction data on the business server

Reply:
Yeah, I'm going to remove it in ANCHOR-439.